sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -265,7 +265,9 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
     }
 
     val statement = plan(ctx.statement)
-    if (isExplainableStatement(statement)) {
+    if (statement == null) {
+      null  // This is enough since ParseException will raise later.
```

Comment:
Where will the parseException be raised? Could you be a bit more specific in your comment?

Maybe add a small test for this? Could you also add a few unit tests (not end-2-end) to SparkSqlParserSuite.